addons/BLUAlert/BLUAlert.lua

```diff
@@ -87,3 +85,14 @@ windower.register_event('action', function(action)
         end
     end
 end)
+
+--Prince Headache's Modifications (Ver 1.1.0):
+
+-- Plays a sound when Blue Magic is learned
+windower.register_event('incoming text', function(original, modified, original_mode, modified_mode, blocked)
+    -- Check if the modified message contains "learns" followed by a Blue Magic spell name
+    local spell_name = modified:match("learns%s+(.+)!")
```

Comment:
I'd also add a `%s+` in front of the `learns` string. Furthermore, this should match the `original` text, not the `modified` one. It's possible some addon modifies what the text looks like and changes the `learns` word (for example, something like: "Spell learned: <name>"), then it would no longer match it. Using `original` should make it work reliably in any case.

Finally, I'd also add a check for the `original_mode`. Otherwise it would trigger even when someone writes something in a party chat, for example.